main_test.go

```diff
@@ -313,6 +317,7 @@ kube_pod_container_resource_limits_memory_bytes{namespace="default",pod="pod0",c
 	sort.Strings(gotFiltered)
 
 	if len(expectedSplit) != len(gotFiltered) {
+		fmt.Printf("%s\n", string(body))
```

Comment:
Debug statement?

Reply:
The statement isn’t very useful as is - if you want to fix the problem you end up copying and pasting then viewing diff.  I’m not going to force it but most people adding something have to print the output anyway

Reply:
I understand and that's certainly fair. We can definitely look into adding a helpful diff statement for it. Also, sounds like a potential beginner-friendly issue that we can raise :).

Reply:
Any reason why we can't incorporate this in the next line to look something like:
```go
t.Fatalf("expected different output length\nfull output:\n%s\n", string(body))
```

Reply:
One disadvantage is that it won't be flush, so you can't cut and paste it.  You have to copy, demargin it, then paste it.  The closest equivalent is the way we do the RBAC generation tests in k/k - the test verifies you match the expected output, but if you don't match, makes it easy to add the expected output.  In those tests we typically left align (again, for ease of consumption).

Reply:
I'll drop it for now.

Reply:
I fine with keeping it, thanks for the explanation!